### PR TITLE
video usb logger fix

### PR DESCRIPTION
**Pull request by rmeertens**
parameters width and height set at runtime. 
removed a useless dependency. 
fixed warnings. 
now logging more data!

### CODE REVIEW

File: conf/modules/video_usb_logger.xml [hunk 1]
```diff
@@ -12,7 +12,6 @@
     <define name="VIDEO_USB_LOGGER_HEIGHTH" value="272" description="Size of the to log images"/>
     <define name="VIDEO_USB_LOGGER_JPEG_WITH_EXIF_HEADER" value="TRUE" description="Whether to store data in the exif header or not"/>
   </doc>
-  <depends>video_rtp_stream</depends>
```
Comment (dewagter) [inline conf/modules/video_usb_logger.xml]: I think it now depends on video_thread and some module that compiles has the jpeg.c

**Comment by fvantienen** [comment]
You made this awesome `pose_history.xml` which is even thread safe.. So it would be nice to use it as it synchronizes the data more accurately. This can then be done in the video capturing thread ;)

**Comment by fvantienen** [comment]
Btw isn't there an option to append all the images to one file(like a video file). I know you can do that for certain encodings.. Maybe also for jpeg?

File: conf/modules/video_usb_logger.xml (hunk 1 above)
Comment (flixr) [inline conf/modules/video_usb_logger.xml]: jep, depends on video_thread, the needed jpeg.c should be added to the sources here directly...

**Comment by rmeertens** [comment]
fixed the dependencies and use the pose history. Will take the video file suggestion as a future feature request suggestion ;)

File: sw/airborne/modules/computer_vision/video_usb_logger.c [hunk 2]
```diff
@@ -35,34 +35,30 @@
...
 static FILE *video_usb_logger = NULL;
-struct image_t img_jpeg;
+struct image_t img_jpeg_global;
+bool created_jpeg = FALSE;
 char foldername[512];
 int shotNumber = 0;
 
-void save_shot(struct image_t *img, struct image_t *img_jpeg)
+void save_shot_on_disk(struct image_t *img, struct image_t *img_jpeg);
+void save_shot_on_disk(struct image_t *img, struct image_t *img_jpeg)
```
Comment (flixr) [inline sw/airborne/modules/computer_vision/video_usb_logger.c]: If it's only an internal function, just make it static... then you also don't need the declaration before it...

File: sw/airborne/modules/computer_vision/video_usb_logger.c [hunk 3]
```diff
@@ -35,34 +35,30 @@
...
 {
 
   // Search for a file where we can write to
   char save_name[128];
 
   sprintf(save_name, "%s/img_%05d.jpg", foldername, shotNumber);
```
Comment (flixr) [inline sw/airborne/modules/computer_vision/video_usb_logger.c]: please never use sprintf and use the save snprintf version instead

**Comment by rmeertens** [comment]
Done.

**Comment by flixr** [comment]
thx, but please fix indentation, codey style...

**Comment by rmeertens** [comment]
done

**Pull request merged**